dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/internal/sanitizer/DdUrlSanitizerTest.kt

```diff
@@ -0,0 +1,157 @@
+package com.datadog.gradle.plugin.internal.sanitizer
+
+import com.datadog.gradle.plugin.Configurator
+import com.nhaarman.mockitokotlin2.doThrow
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import java.lang.RuntimeException
+import java.util.Locale
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.catchThrowable
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class DdUrlSanitizerTest {
+    lateinit var testedUrlSanitizer: DdUrlSanitizer
+
+    @BeforeEach
+    fun `set up`() {
+        testedUrlSanitizer = DdUrlSanitizer()
+    }
+
+    @Test
+    fun `M sanitize a HTTP url W sanitize()`(
+        @StringForgery(regex = "http[s]?://github\\.com:[1-9]{2}/[a-z]+/[a-z0-9_]+\\.git")
+        fakeRemoteUrl: String
+    ) {
+        assertThat(testedUrlSanitizer.sanitize(fakeRemoteUrl)).isEqualTo(fakeRemoteUrl)
+    }
+
+    @Test
+    fun `M sanitize a HTTP url W sanitize() { username and password provided }`(
+        @StringForgery(
+            regex = "http[s]?://[a-z]+:[a-z]+@github\\.com:[1-9]{2}/[a-z]+/[a-z0-9_]+\\.git"
+        )
+        fakeRemoteUrl: String
+    ) {
+        assertThat(testedUrlSanitizer.sanitize(fakeRemoteUrl))
+            .matches("http[s]?://github.com:[1-9]{2}/[a-z]+/[a-z0-9_]+\\.git")
```

Comment:
Technically this assertion is not enough. If the sanitizer changed the port number or the path segments, it would still pass but it won't be correct. 
We can instead do something like: 
```kotlin
    fun `M sanitize a HTTP url W sanitize() { username and password provided }`(
        @StringForgery fakeUserName: String,
        @StringForgery fakePassword: String,
        @StringForgery(regex = "[a-z0-9]+\\.[a-z]{3}" fakeHost: String,
        @IntForgery(1, 99999) fakePort: Int,
        @StringForgery fakeOwner: String,
        @StringForgery fakeRepository: String
    ) {
        // Given
        val fakeRemoteUrl = "http://$fakeUserName:$fakePassword@$fakeHost/$fakeOwner/$fakeRepository.git"

        // When
        val result = testedUrlSanitizer.sanitize(fakeRemoteUrl)

        // Then
        assertThat(result)
            .isEqualTo("http://$fakeHost/$fakeOwner/$fakeRepository.git")
```

And the same comment applies to all the test functions in this class

Reply:
Yes...I do agree, I think it's better to provide separated values for all this parameters.